web/modules/custom/myroute/src/Controller/MyrouteController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Drupal\myroute\Controller;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Controller\ControllerBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a controller for the Myroute module.
+ */
+class MyrouteController extends ControllerBase {
+
+  /**
+   * The account object for the currently logged-in user.
+   *
+   * @var Drupal\Core\Session\AccountInterface
+   */
+  protected $account;
+
+  /**
+   * Constructs a MyrouteController object.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The account object for the currently logged-in user.
+   */
+  public function __construct(AccountInterface $account) {
+    $this->account = $account;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_user')
+    );
+  }
+
+  /**
+   * Displays a message on the practicing route.
+   *
+   * @return array
+   *   A render array containing the markup for the message.
+   */
+  public function practicingRoute() {
+    if ($this->account->hasPermission('view myroute')) {
+      AccessResult::Allowed();
```

Comment:
What are we trying to do here and in line 55?
We are not returning or using these in anyway.
So what's the purpose?

Reply:
Have cross checked again, found that it is blocking the unauthorised users. 
However Updated by adding the return statements.